lib/server.js

```diff
@@ -836,6 +836,10 @@ WalletService.prototype._selectTxInputs = function(txp, cb) {
   self.getUtxos(function(err, utxos) {
     if (err) return cb(err);
 
+    utxos = _.reject(utxos, function(utxo) {
```

Comment:
A more efficient way to filter unwanted utxos would be to first index the `utxos` array and then iterating the `utxosToExclude` array performing a O(1) lookup. I realise this is not such a big issue as these arrays will be rather small but the alternative solution is easy enough IMO.